codalab/model/bundle_model.py

```diff
@@ -520,23 +520,73 @@ def _execute_query(self, query):
             rows = connection.execute(query).fetchall()
         return [row[0] for row in rows]
 
+    # TODO(sckoo): merge with batch_get_bundles_rest when REST migration complete
     def batch_get_bundles(self, **kwargs):
         '''
         Return a list of bundles given a SQLAlchemy clause on the cl_bundle table.
         '''
         clause = self.make_kwargs_clause(cl_bundle, kwargs)
         with self.engine.begin() as connection:
             bundle_rows = connection.execute(
-              cl_bundle.select().where(clause)
+                cl_bundle.select().where(clause)
+            ).fetchall()
+            if not bundle_rows:
+                return []
+            uuids = set(bundle_row.uuid for bundle_row in bundle_rows)
+            dependency_rows = connection.execute(
+                cl_bundle_dependency.select().where(
+                    cl_bundle_dependency.c.child_uuid.in_(uuids)
+                ).order_by(cl_bundle_dependency.c.id)).fetchall()
+            metadata_rows = connection.execute(
+                cl_bundle_metadata.select().where(
+                    cl_bundle_metadata.c.bundle_uuid.in_(uuids)
+                )).fetchall()
+
+        # Make a dictionary for each bundle with both data and metadata.
+        bundle_values = {row.uuid: str_key_dict(row) for row in bundle_rows}
+        for bundle_value in bundle_values.itervalues():
+            bundle_value['dependencies'] = []
+            bundle_value['metadata'] = []
+        for dep_row in dependency_rows:
+            if dep_row.child_uuid not in bundle_values:
+                raise IntegrityError(
+                    'Got dependency %s without bundle' % (dep_row,))
+            bundle_values[dep_row.child_uuid]['dependencies'].append(dep_row)
+        for metadata_row in metadata_rows:
+            if metadata_row.bundle_uuid not in bundle_values:
+                raise IntegrityError(
+                    'Got metadata %s without bundle' % (metadata_row,))
+            bundle_values[metadata_row.bundle_uuid]['metadata'].append(
+                metadata_row)
+
+        # Construct and validate all of the retrieved bundles.
+        sorted_values = sorted(bundle_values.itervalues(),
+                               key=lambda r: r['id'])
+        bundles = [
+            get_bundle_subclass(bundle_value['bundle_type'])(bundle_value)
+            for bundle_value in sorted_values
+            ]
+        for bundle in bundles:
+            bundle.validate()
+        return bundles
+
+    def batch_get_bundles_rest(self, **kwargs):
```

Comment:
Yes the plan would be to delete the newly added batch_get_bundles_rest and keep the old method.